packages/pg-pubsub/src/PgSubscriptionResolverPlugin.ts

```diff
@@ -101,13 +102,41 @@ const PgSubscriptionResolverPlugin: Plugin = function(builder, { pubsub }) {
                 "filter provided to pgSubscription must be a function"
               );
             }
-            return withFilter(() => asyncIterator, filter)(
+            asyncIterator = withFilter(() => asyncIterator, filter)(
               parent,
               args,
               resolveContext,
               resolveInfo
             );
           }
+
+          if (initialEvent) {
+            if (typeof initialEvent !== "function") {
+              throw new Error(
+                "initialEvent provided to pgSubscription must be a function"
+              );
+            }
+
+            return (async function* subscribeWithInitialEvent() {
+              const event = await initialEvent(
+                args,
+                resolveContext,
+                resolveInfo
+              );
+              if (event === null || typeof event !== "object") {
```

Comment:
TypeScript can infer more things if you switch this order.

```suggestion
              if (typeof event !== "object" || event === null) {
```

Reply:
Actually, I think returning `null` or `undefined` should be entirely valid - we just auto-skip it.

```js
// Only send initialEvent when return value is not null/undefined.
if (event != null) {
  if (typeof event !== "object") {
    throw ...
  }
  yield { ...event, topic };
}
for await ...
```

Reply:
Didn't know about the TS trick, nice! Regarding the empty event, I agree.